obs/api/app/controllers/api/storage.py

```diff
@@ -302,38 +301,33 @@ def post(self, bucket_name):
         args = parser.parse_args()
         secret_key = args["secret_key"].replace(" ", "+")
 
-        with tempfile.TemporaryDirectory() as tempdir:
-            file = request.files["files"]
-            filename = secure_filename(file.filename)
-            object_name = args["object_name"] if args["object_name"] else filename
-            filename = os.path.join(tempdir, filename)
-            file.save(filename)
+        file = request.files["files"]
+        filename = secure_filename(file.filename)
```

Comment:
its okey to use secure_filename rather than guess mimetypes, try set allowed extension to avoid from injecting script, as my suggestion check https://flask.palletsprojects.com/en/1.1.x/patterns/fileuploads/

Reply:
in s3 object key use content_type instead file extension to identify the file
https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-keys

and in upload function file stored in FileStorage and will be erased as soon as the request finish.

Reply:
oh i see....